From andrew@lunn.ch  Fri Dec 18 10:28:07 2009
From: Andrew Lunn <andrew@lunn.ch>
Date: Sat, 12 Dec 2009 23:39:44 +0100
Subject: Staging: batman-adv: Always receive when discarding.
To: gregkh@suse.de
Cc: devel@linuxdriverproject.org, Andrew Lunn <andrew@lunn.ch>, Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Message-ID: <1260657587-1135-4-git-send-email-andrew@lunn.ch>


It does not matter if the interface is to be activated or not, we must
read the packet in order that it be discarded.

Signed-off-by: Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Signed-off-by: Andrew Lunn <andrew@lunn.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/batman-adv/routing.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/staging/batman-adv/routing.c
+++ b/drivers/staging/batman-adv/routing.c
@@ -1177,8 +1177,7 @@ static int discard_one_packet(struct bat
 {
 	int result = -EAGAIN;
 
-	if ((batman_if->if_active == IF_TO_BE_ACTIVATED)
-		&& (batman_if->raw_sock)) {
+	if (batman_if->raw_sock) {
 			result = receive_raw_packet(batman_if->raw_sock,
 						    packet_buff,
 						    PACKBUFF_SIZE);
